src/modules/job-manager/submit.c

```diff
@@ -107,19 +107,41 @@ zlist_t *submit_enqueue_jobs (struct queue *queue, json_t *jobs)
  */
 int submit_post_event (struct event_ctx *event_ctx, struct job *job)
 {
-        char event[64];
-        (void)snprintf (event, sizeof (event),
-                        "%.6f submit userid=%lu priority=%d flags=%d\n",
-                        job->t_submit,
-                        (unsigned long)job->userid,
-                        job->priority,
-                        job->flags);
-        if (event_job_update (job, event) < 0)
-            return -1;
-        if (event_job_action (event_ctx, job) < 0)
-            return -1;
-        return 0;
-};
+    char event[FLUX_KVS_MAX_EVENT_CONTEXT + 1];
+    json_t *o = NULL;
+    char *str = NULL;
+    int save_errno, rv = -1;
+
+    if (!(o = json_pack ("{ s:i s:i s:i }",
```

Comment:
Suggestion: move event encoding to a separate function and just call `flux_kvs_event_encode_timestamp()` rather than encoding manually.